### PR TITLE
Change 'ignore' to 'reload' for cycle point cli opts

### PR DESCRIPTION
**Pull request by MetRonnie**
These changes partially address #4062 

> The name `ignore` [in the `--*-cycle-point=ignore` cli options] is potentially misleading, as one might get the idea it will basically set the final cycle point to `None`. 

- Change `ignore` to `reload`, as we are actually reloading from the `flow.cylc` file

Also

> It looks like `--startcp=ignore` doesn't actually do anything? The start point for a restart is whatever the last incomplete cycle point at shutdown was.

- Disallow `--icp=reload` and `--startcp=reload` because they weren't doing anything

**Requirements check-list**
- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Applied any dependency changes to both `setup.py` and
  `conda-environment.yml`.
- [x] Already covered by existing tests.
- [x] Appropriate change log entry included.
- [ ] (master branch) I have opened a documentation PR at cylc/cylc-doc/pull/XXXX.

### CODE REVIEW

File: tests/functional/restart/50-ignore-stop-point.t [hunk 1]
```diff
@@ -1,77 +0,0 @@
...
-#
-# This program is distributed in the hope that it will be useful,
-# but WITHOUT ANY WARRANTY; without even the implied warranty of
-# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-# GNU General Public License for more details.
-#
-# You should have received a copy of the GNU General Public License
-# along with this program.  If not, see <http://www.gnu.org/licenses/>.
-#-------------------------------------------------------------------------------
-# Test restart with ignore stop point
```
Comment (MetRonnie) [inline tests/functional/restart/50-ignore-stop-point.t:18]: Merged this test with `08-stop-after-cycle-point.t `

**Comment by oliver-sanders** [comment]
> Change ignore to refresh, as we are actually refreshing from the flow.cylc file

Perhaps "reload" would make more sense (matches `cylc reload`).

**Comment by MetRonnie** [comment]
> Perhaps "reload" would make more sense (matches `cylc reload`).

Happy to, @hjoliver do you agree?

**Comment by hjoliver** [comment]
> > Perhaps "reload" would make more sense (matches `cylc reload`).
> 
> Happy to, @hjoliver do you agree?

Yes, agreed

File: cylc/flow/scheduler.py [hunk 2]
```diff
@@ -1844,35 +1844,39 @@ def _check_startup_opts(self) -> None:
...
-        else:
-            for opt in ('icp', 'fcp', 'startcp', 'stopcp', 'starttask'):
-                if getattr(self.options, opt, None) == 'ignore':
-                    raise SchedulerError(
-                        f"option --{opt}=ignore is only valid for restart.")
+        for opt in ('icp', 'startcp', 'starttask'):
+            value = getattr(self.options, opt, None)
+            if self.is_restart:
+                if value is not None:
+                    raise UserInputError(
```
Comment (oliver-sanders) [inline cylc/flow/scheduler.py:1853]: Checked that this change of exception doesn't mess with the error catching.

Works ok, changes the exit code from `1` to `3` but that doesn't matter.

Error message changes from:

```
2021-08-05T10:39:28+01:00 ERROR - Workflow shutting down - option --startcp=reload is not valid (only --fcp and --stopcp can be 'reload')
```

To:

```
2021-08-05T10:38:53+01:00 ERROR - Workflow shutting down - UserInputError: option --startcp=reload is not valid 
```

**Review by oliver-sanders — APPROVED** [review]

**Review by hjoliver — APPROVED** [review]
:+1:

**Pull request merged**